File: Lv2/기능개발.kt
class Solution {
    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
        val tmp = mutableListOf<Int>()
        for (i in progresses.indices) {
            if ((100 - progresses[i]) % speeds[i] != 0) {
                tmp.add((100 - progresses[i]) / speeds[i] + 1)
            } else {
                tmp.add((100 - progresses[i]) / speeds[i])
            }
        }
        var count = 0
        var flag = 0
        var answer = mutableListOf<Int>()
        for (i in tmp) {
            if (flag == 0) {
                flag = i
                count++
            } else {
                if (flag >= i) {
                    count++
                } else {
                    flag = i
                    answer.add(count)
                    count = 1
                }
            }
        }
        answer.add(count)
        return answer.toIntArray()
    }
}

File: Lv2/기능개발.py
from queue import deque

def solution(progresses, speeds):
    answer = []
    tmp = deque(progresses)
    ref = deque(speeds)
    while len(tmp) != 0:
        count = 0
        for i in range(len(tmp)):
            tmp[i] += ref[i]
        while len(tmp) != 0 and tmp[0] >= 100:
            count += 1
            tmp.popleft()
            ref.popleft()
        if count!=0:
            answer.append(count)
    return answer